### DISCUSSION TITLE
How to use tsfresh with unsupervised learning

### DISCUSSION THREAD

**Discussion by chauhansaurabhb** [blue-yonder/tsfresh#859] [Q&A] [answered]
Thanks for sharing this library.

I am looking to use this library with reference to unsupervised learning. I have one curve (time ~ value) and I have only three columns in dataset i.e. `id, time, value` (here I have day wise sum of values i.e. aggregated values) and would like to use this library to extract features followed by clustering techniques.

However, I can able to extract features but when I select optimal features using `features_filtered = select_features(extracted_features, y)` but how can I pass value of y? I don't have label column in my dataset.

Could you please elaborate more on "How to use tsfresh library for unsupervised learning?"

Thanking you!

**Comment by kempa-liehr** [comment, marked as answer]
Hi chauhansaurabhb,
Thanks for your question.
The feature selection functionality of tsfresh only works for supervised learning. As  you correctly noted: You do not have y values.
Cheers, Andreas

**Comment by chauhansaurabhb** [comment]
Thank you for the clarification!

**Comment by kempa-liehr** [comment]
@chauhansaurabhb: We have recently published a paper on unsupervised feature selection, which might be useful for your application: https://doi.org/10.1109/JSEN.2021.3084970
